de.metas.business/src/main/java/de/metas/pricing/service/impl/PriceListBL.java

```diff
@@ -44,41 +44,55 @@
 import de.metas.util.Services;
 import lombok.NonNull;
 
-@SuppressWarnings("unused")
 public class PriceListBL implements IPriceListBL
 {
-
 	private final IPriceListDAO priceListDAO = Services.get(IPriceListDAO.class);
+	private final ICurrencyDAO currenciesRepo = Services.get(ICurrencyDAO.class);
 
 	@Override
 	public CurrencyPrecision getPricePrecision(@NonNull final PriceListId priceListId)
 	{
 		final I_M_PriceList priceList = priceListDAO.getById(priceListId);
+		return getPricePrecision(priceList);
+	}
+
+	private CurrencyPrecision getPricePrecision(final I_M_PriceList priceList)
+	{
 		return CurrencyPrecision.ofInt(priceList.getPricePrecision());
 	}
 
 	@Override
 	public CurrencyPrecision getAmountPrecision(@NonNull final PriceListId priceListId)
 	{
 		final I_M_PriceList priceList = priceListDAO.getById(priceListId);
+
+		final CurrencyPrecision taxPrecision = getTaxPrecision(priceList);
+
 		if (priceList.isRoundNetAmountToCurrencyPrecision())
 		{
-			return Services.get(ICurrencyDAO.class).getStdPrecision(CurrencyId.ofRepoId(priceList.getC_Currency_ID()));
+			return taxPrecision;
 		}
 		else
 		{
-			return CurrencyPrecision.ofInt(priceList.getPricePrecision());
+			final CurrencyPrecision pricePrecision = getPricePrecision(priceList);
+			return pricePrecision.min(taxPrecision);
```

Comment:
aah, this is the fix, right?

Reply:
@metas-ts  right. do u think it's OK for https://github.com/metasfresh/metasfresh/issues/5234 ?